supervisor/mounts/mount.py

```diff
@@ -196,8 +181,11 @@ async def load(self) -> None:
         if not await self.is_mounted():
             await self.reload()
 
-    async def _update_state(self) -> UnitActiveState | None:
+    async def _update_state(self) -> None:
         """Update mount unit state."""
+        if not self.unit:
```

Comment:
We should add a log here, debug level if expected, warning or critical level if this is unexpected.

Reply:
I can add a debug log if we want. This is really just type check stuff as it can't actually occur. All the places that call this private method first attempt to update the unit info and fail out if they can't.